templates/php/src/Services/Service.php.twig

```diff
@@ -58,43 +59,74 @@ class {{ service.name | caseUcfirst }} extends Service
-{% for param in method.parameters.header %}
-                '{{ param.name }}' => ${{ param.name | caseCamel }},
-{% endfor %}
-{% for key, header in method.headers %}
-                '{{ key }}' => '{{ header }}',
-{% endfor %}
-                ], $params);
+        $size = 0;
+        $mimeType = null;
+        $postedName = null;
+        if(empty(${{ parameter.name | caseCamel }}->getPath())) {
+            $size = strlen(${{ parameter.name | caseCamel }}->getData());
+            $mimeType = ${{ parameter.name | caseCamel }}->getMimeType();
+            $postedName = ${{ parameter.name | caseCamel }}->getFilename();
+            if ($size <= Client::CHUNK_SIZE) {
+                $params['{{ parameter.name | caseCamel }}'] = new \CURLFile('data://' . $mimeType . ';base64,' . base64_encode(${{ parameter.name | caseCamel }}->getData()), $mimeType, $postedName);
+                return $this->client->call(Client::METHOD_POST, $path, [
+    {% for param in method.parameters.header %}
+                    '{{ param.name }}' => ${{ param.name | caseCamel }},
+    {% endfor %}
+    {% for key, header in method.headers %}
+                    '{{ key }}' => '{{ header }}',
+    {% endfor %}
+                    ], $params);
+            }
+        } else {
+            $size = filesize(${{ parameter.name | caseCamel }}->getPath());
+            $mimeType = ${{ parameter.name | caseCamel }}->getMimeType() ?? mime_content_type(${{ parameter.name | caseCamel }}->getPath());
+            $postedName = ${{ parameter.name | caseCamel }}->getFilename() ?? basename(${{ parameter.name | caseCamel }}->getPath());
+            //send single file if size is less than or equal to 5MB
+            if ($size <= Client::CHUNK_SIZE) {
+                $params['{{ parameter.name }}'] = new \CURLFile(${{ parameter.name | caseCamel }}->getPath(), $mimeType, $postedName);
+                return $this->client->call(Client::METHOD_{{ method.method | caseUpper }}, $path, [
+    {% for param in method.parameters.header %}
+                    '{{ param.name }}' => ${{ param.name | caseCamel }},
+    {% endfor %}
+    {% for key, header in method.headers %}
+                    '{{ key }}' => '{{ header }}',
+    {% endfor %}
+                    ], $params);
+            }
         }
 
         $id = '';
         $counter = 0;
 
 {% for parameter in method.parameters.all %}
 {% if parameter.isUploadID %}
-            if(${{ parameter.name | caseCamel | escapeKeyword }} != 'unique()') {
-                try {
-                    $response = $this->client->call(Client::METHOD_GET, new URL($path . '/' . {{ parameter.name }}), headers);
-                    $counter = $response['chunksUploaded'] ?? 0;
-                } catch(\Exception $e) {
-                }
+        if(${{ parameter.name | caseCamel | escapeKeyword }} != 'unique()') {
+            try {
+                $response = $this->client->call(Client::METHOD_GET, $path . '/' . ${{ parameter.name }},);
```

Comment:
You removed headers here, and kept comma. Looks like syntax error, also unwanted change probably.